shell.nix

```diff
@@ -1 +1,12 @@
-import ./. { makeEnv = true; }
+{pkgs ? import ./nixpkgs.nix {}}:
+
+with pkgs;
+
+mkShell {
+  LANG="C.UTF-8";
+
+  buildInputs = [
+    ghc
+    ghcid
```

Comment:
No z3 in the inputs?

Have you tested the setup with `nix-shell --pure`?

I'd leave `ghcid` out as it is non-essential to building. But I don't feel strong about it.

Also, I'd prefer to be explicit with the `ghc` compiler version and use instead `haskell.compiler.ghc8107` as input, just so it is easier to guess by looking at the configuration.

Reply:
> Have you tested the setup with `nix-shell --pure`?

As a matter of personal preference, I use project-specific Nix shells as "additions" to my normal tools, which include nonstandard things like `rg`, `fd`, `zoxide`, `fzf`, etc. So I don't myself use `nix-shell --pure`.

The more relevant question is: what would you like me to test inside `nix-shell --pure`?

I have listed the commands I tested in the description. The declarations in `shell-stack.nix` bring `z3` into scope for `stack exec fixpoint -- --solver=z3` to succeed.

What does not work in the current setup is:

```
$ stack install
$ fixpoint tests/pos/adt.fq
Liquid-Fixpoint Copyright 2013-21 Regents of the University of California.
All Rights Reserved.

Working   0% [.................................................................]
fixpoint: fd:13: hGetLine: end of file
```

This is fixed by adding `z3` to `buildInputs` in `shell.nix`.

Is `stack install && fixpoint ...` something that should be supported in addition to `stack exec fixpoint -- ...`? It sounds to me like you're suggesting it should, so I'll add `z3` and extend the description.

---

> I'd leave `ghcid` out as it is non-essential to building. But I don't feel strong about it.

Done.

---

Also, I'd prefer to be explicit with the `ghc` compiler version and use instead `haskell.compiler.ghc8107` as input, just so it is easier to guess by looking at the configuration.

Makes sense, done.

Reply:
(Note that I have updated the description to list the commands that work now.)